### PR TITLE
os

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Open-Source-Teil des Open-Core-Modells umsetzen: Repository](https://trello.com/c/7d6O4x4t/591-open-source-teil-des-open-core-modells-umsetzen-repository)

File: services/src/contexts/simple_context.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+use super::{Context, Db, SimpleSession};
+
+use async_trait::async_trait;
+use tokio::sync::{RwLockReadGuard, RwLockWriteGuard};
+
+#[async_trait]
+pub trait SimpleContext: Context<Session = SimpleSession> {
+    // async fn default_session(&self) -> &SimpleSession;
```
Comment (michaelmattig) [inline services/src/contexts/simple_context.rs:8]: remove commented out lines

File: services/src/contexts/mod.rs [hunk 2]
```diff
@@ -63,34 +56,58 @@ pub trait Context: 'static + Send + Sync + Clone {
 
     fn query_context(&self) -> Result<Self::QueryContext>;
 
-    fn execution_context(&self, session: &Session) -> Result<Self::ExecutionContext>;
+    fn execution_context(&self, session: Self::Session) -> Result<Self::ExecutionContext>;
+
+    async fn session_id_to_session(&self, session_id: SessionId) -> Result<Self::Session>;
```
Comment (michaelmattig) [inline services/src/contexts/mod.rs:61]: session_by_id() oder session(id)

File: services/src/pro/handlers/users.rs [hunk 3]
```diff
@@ -0,0 +1,767 @@
...
+use crate::projects::STRectangle;
+use crate::util::user_input::UserInput;
+use crate::util::IdResponse;
+
+use snafu::ResultExt;
+use uuid::Uuid;
+use warp::reply::Reply;
+use warp::Filter;
+
+// TODO: move to pro, create session-handler
```
Comment (michaelmattig) [inline services/src/pro/handlers/users.rs:19]: remove todo

File: services/src/pro/users/session.rs [hunk 4]
```diff
@@ -18,7 +16,7 @@ pub struct UserInfo {
 
 #[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
 #[serde(rename_all = "camelCase")]
-pub struct Session {
+pub struct UserSession {
```
Comment (michaelmattig) [inline services/src/pro/users/session.rs:19]: ProSession? analogue to ProContext

File: services/src/pro/datasets/in_memory.rs [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+
+impl ProHashMapStorable for GdalMetaDataStatic {
+    fn store(&self, id: InternalDatasetId, db: &mut ProHashMapDatasetDb) -> TypedResultDescriptor {
+        db.gdal_datasets.insert(id, Box::new(self.clone()));
+        self.result_descriptor.clone().into()
+    }
+}
+
+#[async_trait]
+impl DatasetStore<UserSession> for ProHashMapDatasetDb {
```
Comment (michaelmattig) [inline services/src/pro/datasets/in_memory.rs:118]: `typedef ProDatasetStore = DatasetStore<UserSession>`? also for other DBS?

File: services/src/projects/hashmap_projectdb.rs [hunk 6]
```diff
@@ -242,112 +146,13 @@ mod test {
...
-        let projects = project_db.list(user, options).await.unwrap();
-        assert!(projects[0].name == "Own");
-        assert_eq!(projects.len(), 1);
-    }
-
     #[tokio::test]
     async fn list() {
         let mut project_db = HashMapProjectDb::default();
-        let user = UserId::new();
+        let session = SimpleSession::default();
```
Comment (michaelmattig) [inline services/src/projects/hashmap_projectdb.rs:155]: always use default() or mock()

File: services/src/projects/project.rs [hunk 7]
```diff
@@ -62,7 +61,7 @@ impl Project {
     /// If the updates layer list is longer than the current list,
     /// it just inserts new layers to the end.
     ///
-    pub fn update_project(&self, update: UpdateProject, user: UserId) -> Result<Project> {
+    pub fn update_project(&self, update: UpdateProject) -> Result<Project> {
```
Comment (michaelmattig) [inline services/src/projects/project.rs:64]: project versioning only in pro?

File: services/src/server.rs [hunk 8]
```diff
@@ -70,20 +64,6 @@ pub async fn start_server(
...
-                        &get_config_element::<config::Postgres>()?.config_string,
-                    )?,
-                    NoTls,
-                )
-                .await?;
-
-                start(shutdown_rx, static_files_dir, bind_address, ctx).await
-            }
-            #[cfg(not(feature = "postgres"))]
             panic!("Postgres backend was selected but the postgres feature wasn't activated during compilation")
```
Comment (michaelmattig) [inline services/src/server.rs:67]: say postgres is pro only? do we need two enums for pro/free backends?

File: LICENSE [hunk 9]
```diff
@@ -0,0 +1,21 @@
+Copyright (c) 2021-present Geo Engine GmbH
+
+Portions of this software are licensed as follows:
+
+* All content that resides under the "datatypes/pro/" directory of this repository, if that directory exists, is licensed under the license defined in "datatypes/pro/LICENSE".
```
Comment (michaelmattig) [inline LICENSE:5]: is the slash at the end of the directories necessary?

**Review by michaelmattig — CHANGES_REQUESTED** [review]
.

**Comment by ChristianBeilschmidt** [comment]
> just call it session, there is only the default one

File: services/src/pro/users/session.rs (hunk 4 above)
Comment (ChristianBeilschmidt) [reply services/src/pro/users/session.rs:19]: lassen wir so

File: services/src/pro/datasets/in_memory.rs (hunk 5 above)
Comment (ChristianBeilschmidt) [reply services/src/pro/datasets/in_memory.rs:118]: #later

**Pull request merged**